File: yandex_algorithms_training5/hw1/go/2_football_commentator.go
package main

import (
	"fmt"
	"strconv"
	"strings"
)

func countGoals(first []int, second []int, hoa int) int {
	var firstTeamHome, firstTeamAway int
	var secondTeamHome, secondTeamAway int

	if hoa == 1 {
		firstTeamHome = first[0]
		firstTeamAway = second[0]
		secondTeamHome = second[1]
		secondTeamAway = first[1]
	} else {
		firstTeamHome = second[0]
		firstTeamAway = first[0]
		secondTeamHome = first[1]
		secondTeamAway = second[1]
	}

	firstTeamGoals := firstTeamHome + firstTeamAway
	secondTeamGoals := secondTeamHome + secondTeamAway

	if firstTeamGoals > secondTeamGoals {
		return 0
	}

	if firstTeamGoals == 0 && secondTeamGoals == 0 {
		return 1
	}

	equalizeGoals := secondTeamGoals - firstTeamGoals

	if hoa == 1 {
		if equalizeGoals+firstTeamAway > secondTeamAway {
			return equalizeGoals
		} else {
			return equalizeGoals + 1
		}
	} else {
		if firstTeamAway > secondTeamAway {
			return equalizeGoals
		} else {
			return equalizeGoals + 1
		}
	}
}

func main() {
	var firstGame, secondGame string
	var homeOrAway int

	fmt.Scan(&firstGame, &secondGame, &homeOrAway)

	firstGameScores := strings.Split(firstGame, ":")
	secondGameScores := strings.Split(secondGame, ":")

	var firstGameScoresInt, secondGameScoresInt []int

	for _, value := range firstGameScores {
		num, err := strconv.Atoi(value)
		if err != nil {
			panic(err)
		}

		firstGameScoresInt = append(firstGameScoresInt, num)
	}

	for _, value := range secondGameScores {
		num, err := strconv.Atoi(value)
		if err != nil {
			panic(err)
		}

		secondGameScoresInt = append(secondGameScoresInt, num)
	}

	fmt.Println(countGoals(firstGameScoresInt, secondGameScoresInt, homeOrAway))
}

File: yandex_algorithms_training5/hw1/python/2_football_commentator.py
score1 = input().split(':')
score2 = input().split(':')

home_or_away_loc = int(input())

score1 = [int(goal) for goal in score1]
score2 = [int(goal) for goal in score2]


def count_goals(first: list, second: list, home_or_away: int) -> int:
    if home_or_away == 1:
        first_team_home = first[0]
        first_team_away = second[0]
        second_team_home = second[1]
        second_team_away = first[1]
    else:
        first_team_home = second[0]
        first_team_away = first[0]
        second_team_home = first[1]
        second_team_away = second[1]

    first_team_goals = first_team_home + first_team_away
    second_team_goals = second_team_home + second_team_away

    if first_team_goals > second_team_goals:
        return 0

    if first_team_goals == second_team_goals == 0:
        return 1

    equalize_goals = second_team_goals - first_team_goals

    if home_or_away == 1:
        if equalize_goals + first_team_away > second_team_away:
            return equalize_goals
        else:
            return equalize_goals + 1
    else:
        if first_team_away > second_team_away:
            return equalize_goals
        else:
            return equalize_goals + 1


print(count_goals(score1, score2, home_or_away_loc))
